### ISSUE TITLE
firebase_auth/cloud_firestore plugins: Incompatible plugins firebase_auth and cloud_firestore

### ISSUE THREAD

**Issue by naveenaa** [FirebaseExtended/flutterfire#1536]
**Describe the bug**
Android, Flutter/Firebase/cloud_firestore plugin version incompatibleresults in errors
Issue detailed in
https://github.com/flutter/website/issues/3333

**To Reproduce**
set the following versions in pubspec.yaml
Page URL: https://flutter.dev/docs/development/data-and-backend/firebase.html
Page source: https://github.com/flutter/website/tree/master/src/docs/development/data-and-backend/firebase.md

Found a typo? You can fix it yourself by going to the page source and clicking the pencil icon. Or finish creating this issue.

PLEASE DEFINE FOR FLUTTER APP, VERSIONS TO USE IF firebase_auth credential verification has to be executed. THANKS!
firebase_core: ^0.4.0+9
firebase_auth: ^0.8.0
cloud_firestore: ^0.7.0+1

Description of issue:
Versioning incompatible errors:
firebase_core: ^0.4.0+9
firebase_auth: ^0.8.0
cloud_firestore: ^0.7.0+1

depends on cloud_firestore ^0.7.0+1 which depends on firebase_core ^0.2.2, firebase_core ^0.2.2 is required.
So, because programApp depends on firebase_core ^0.4.0+9, version solving failed.
pub get failed (1)
Process finished with exit code 1

Tried changing to:
based on https://codelabs.developers.google.com/codelabs/flutter-firebase/#3
dependencies:
flutter:
sdk: flutter
cloud_firestore: ^0.11.0+1 # new

And observe more errors:
Because firebase_auth >=0.8.0+1 <0.10.0 depends on firebase_core ^0.3.0 and no versions of firebase_auth match >0.8.0 <=0.8.0, firebase_auth >0.8.0 <0.10.0 requires firebase_core ^0.3.0.
And because firebase_auth 0.8.0 depends on firebase_core ^0.2.5+1, firebase_auth >=0.8.0 <0.10.0 requires firebase_core ^0.2.5+1 or ^0.3.0.
So, because programApp depends on both firebase_core ^0.4.0+9 and firebase_auth ^0.8.0, version solving failed.
pub get failed (1)
Process finished with exit code 1
**Expected behavior**

**Additional context**

firebase_auth: ^0.15.0+1
cloud_firestore: ^0.12.9+5

removed firebase_core
observe following errors
```
Launching lib/main.dart on Redmi Note 6 Pro in debug mode...
Initializing gradle...
Resolving dependencies...
Running Gradle task 'assembleDebug'...
registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)

FAILURE: Build failed with an exception.

What went wrong:
Execution failed for task ':app:preDebugBuild'.
Android dependency 'android.arch.lifecycle:runtime' has different version for the compile (1.1.0) and runtime (1.1.1) classpath. You should manually set the same version via DependencyResolution

Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

Get more help at https://help.gradle.org

BUILD FAILED in 15s
Finished with error: Gradle task assembleDebug failed with exit code 1
```

If the version number changed to:
firebase_auth: 0.11.1+12

observe code errors of the plugin
```
final PhoneVerificationCompleted verificationCompleted =
(AuthCredential phoneAuthCredential) async {

  AuthResult authResult = await _auth.signInWithCredential(phoneAuthCredential);
```

<img width="658" alt="Screen Shot 2019-12-03 at 3 10 14 PM" src="https://user-images.githubusercontent.com/8305626/70039321-b9f5b780-15df-11ea-80bd-bac968466f21.png">

**Comment by naveenaa** [comment]
Tried the following changes: build.gradle as specified in
https://firebase.google.com/docs/flutter/setup?platform=android
<img width="1046" alt="Screen Shot 2019-12-03 at 4 20 10 PM" src="https://user-images.githubusercontent.com/8305626/70044635-bfa3cb00-15e8-11ea-9253-a51f80448276.png">

dependencies {
        classpath 'com.android.tools.build:gradle:3.2.1'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
        //classpath 'com.google.gms:google-services:3.2.1'
        classpath 'com.google.gms:google-services:4.3.3'
        //classpath 'com.google.gms:google-services:4.3.0'
    }

**Comment by naveenaa** [comment]
* What went wrong:
Execution failed for task ':app:preDebugBuild'.
> Android dependency 'android.arch.lifecycle:runtime' has different version for the compile (1.1.0) and runtime (1.1.1) classpath. You should manually set the same version via DependencyResolution

**Comment by naveenaa** [comment]
It will be helpful if using firebase_auth plugin, the dependencies related to cloud_firestore and phoneVerification are detailed

Thank you!

**Comment by naveenaa** [comment]
issue detailed here https://github.com/FirebaseExtended/flutterfire/issues/1103

**Comment by Salakar** [comment]
Hey, as part of our on-going work for #2582, this has been resolved in our Firebase Firestore rework (#2913) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
